server/index.js

```diff
@@ -259,6 +259,32 @@ app.get('/api/user/listings/:listingId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.delete('/api/user/listings/:listingId', (req, res, next) => {
+  const listingId = Number(req.params.listingId);
+  if (!Number.isInteger(listingId) || listingId <= 0) {
+    throw new ClientError(400, '"listingId" must be a positive integer.');
+  }
+  const sql = `
+    delete from "listings"
+    where "listingId" = $1
+    returning *;
+    `;
+  const values = [listingId];
+  db.query(sql, values)
+    .then(result => {
+      const listing = result.rows[0];
+      if (!listing) {
+        throw new ClientError(400, `Cannot find listing with "listingId: " ${listingId}`);
+      } else {
+        res.sendStatus(204);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      throw new ClientError(500, 'An unexpected error occurred.');
+    });
```

Comment:
Rather than throw a 500, pass your `err` to `next()` so that your error handling middleware deals with it for you.